### ISSUE TITLE
[Bug]: E2EE Generated CSR isn't valid

### ISSUE THREAD

**Issue by deajan** [nextcloud/desktop#5910]
### ⚠️ Before submitting, please verify the following: ⚠️

- [X] This is a **bug**, not a question or a configuration issue.
- [X] This issue is **not** already reported on Github (I've searched it).
- [X] Nextcloud Server and Desktop Client are **up to date**. See [Server Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) and [Desktop Releases](https://nextcloud.com/install/#install-clients) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/)

### Bug description

When trying to enable E2EE, nothing happens client side.
On the server, there are logs that the CSR is incorrect.

I've documented everything at the following issue: https://github.com/nextcloud/end_to_end_encryption/issues/424

Since I now think that the problem comes from the nc desktop client, I'm opening this issue.
[debug_arch.zip](https://github.com/nextcloud/desktop/files/12113348/debug_arch.zip)

### Steps to reproduce

1. Install fresh NC 27 server and NC desktop 3.9.0 on win10 x64
2. Click on "Enable encryption"
3. Nothing happens client side
4. Server will have logs in nextcloud.log stating the CSR is invalid.

### Expected behavior

Proper CSR sent to server, which signs CSR with it's private key and sends certificate back.

### Which files are affected by this bug

<none>

### Operating system

Windows

### Which version of the operating system you are running.

Windows 10 22H2

### Package

Other

### Nextcloud Server version

27.0.1.2

### Nextcloud Desktop Client version

3.9.0

### Is this bug present after an update or on a fresh install?

Fresh desktop client install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### Are you using an external user-backend?

- [X] Default internal user-backend
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Nextcloud Server logs

```shell
{
    "reqId": "ZKgUt88faq7AiJB5It7fsQAAAM8",
    "level": 3,
    "time": "2023-07-07T13:35:51+00:00",
    "remoteAddr": "::ffff:XX.XX.XX.XX",
    "user": "me@example.tld",
    "app": "PHP",
    "method": "POST",
    "url": "/ocs/v2.php/apps/end_to_end_encryption/api/v1/public-key?format=json",
    "message": "openssl_csr_sign(): Signature did not match the certificate request at /var/www/cloud.example.tld/ftp/www/apps/end_to_end_encryption/lib/SignatureHandler.php#53",
    "userAgent": "Mozilla/5.0 (Windows) mirall/3.9.0stable-Win64 (build 20230613) (Nextcloud, windows-10.0.19045 ClientArchitecture: x86_64 OsArchitecture: x86_64)",
    "version": "27.0.0.8",
    "data": {
        "app": "PHP"
    }
}{
    "reqId": "ZKgUt88faq7AiJB5It7fsQAAAM8",
    "level": 3,
    "time": "2023-07-07T13:35:51+00:00",
    "remoteAddr": "::ffff:XX.XX.XX.XX",
    "user": "me@example.tld",
    "app": "end_to_end_encryption",
    "method": "POST",
    "url": "/ocs/v2.php/apps/end_to_end_encryption/api/v1/public-key?format=json",
    "message": "could not sign the CSR, please make sure to submit a valid CSR",
    "userAgent": "Mozilla/5.0 (Windows) mirall/3.9.0stable-Win64 (build 20230613) (Nextcloud, windows-10.0.19045 ClientArchitecture: x86_64 OsArchitecture: x86_64)",
    "version": "27.0.0.8",
    "exception": {
        "Exception": "BadMethodCallException",
        "Message": "could not sign the CSR, please make sure to submit a valid CSR",
        "Code": 0,
        "Trace": [{
                "file": "/var/www/cloud.example.tld/ftp/www/apps/end_to_end_encryption/lib/Controller/KeyController.php",
                "line": 189,
                "function": "sign",
                "class": "OCA\\EndToEndEncryption\\SignatureHandler",
                "type": "->"
            }, {
                "file": "/var/www/cloud.example.tld/ftp/www/lib/private/AppFramework/Http/Dispatcher.php",
                "line": 230,
                "function": "createPublicKey",
                "class": "OCA\\EndToEndEncryption\\Controller\\KeyController",
                "type": "->",
                "args": ["*** sensitive parameters replaced ***"]
            }, {
                "file": "/var/www/cloud.example.tld/ftp/www/lib/private/AppFramework/Http/Dispatcher.php",
                "line": 137,
                "function": "executeController",
                "class": "OC\\AppFramework\\Http\\Dispatcher",
                "type": "->"
            }, {
                "file": "/var/www/cloud.example.tld/ftp/www/lib/private/AppFramework/App.php",
                "line": 183,
                "function": "dispatch",
                "class": "OC\\AppFramework\\Http\\Dispatcher",
                "type": "->"
            }, {
                "file": "/var/www/cloud.example.tld/ftp/www/lib/private/Route/Router.php",
                "line": 315,
                "function": "main",
                "class": "OC\\AppFramework\\App",
                "type": "::"
            }, {
                "file": "/var/www/cloud.example.tld/ftp/www/ocs/v1.php",
                "line": 64,
                "function": "match",
                "class": "OC\\Route\\Router",
                "type": "->"
            }, {
                "file": "/var/www/cloud.example.tld/ftp/www/ocs/v2.php",
                "line": 23,
                "args": ["/var/www/cloud.example.tld/ftp/www/ocs/v1.php"],
                "function": "require_once"
            }
        ],
        "File": "/var/www/cloud.example.tld/ftp/www/apps/end_to_end_encryption/lib/SignatureHandler.php",
        "Line": 55,
        "message": "could not sign the CSR, please make sure to submit a valid CSR",
        "exception": {},
        "CustomMessage": "could not sign the CSR, please make sure to submit a valid CSR"
    }
```

### Additional info

Willing to make additional tests if requested.

**Comment by deajan** [comment]
Same error still present on desktop app 3.9.2.

**Comment by deajan** [comment]
Closed in favor of #5989
